primefaces-integration-tests/src/main/java/org/primefaces/integrationtests/fileupload/FileUploadView.java

```diff
@@ -58,9 +66,43 @@ public void doFileUpload(UploadedFile uf) {
     private void readFile(UploadedFile uf) throws IOException {
         byte[] buf = new byte[(int) uf.getSize() + 1];
         int len = uf.getInputStream().read(buf);
-        if (len != (int) uf.getSize()) {
+        if (Math.max(0, len) != (int) uf.getSize()) {
             throw new IllegalStateException("unexpected file size " + uf.getSize() + " <> " + len);
         }
+        if (!isAllowedType(uf.getFileName())) {
+            throw new IllegalStateException("unexpected file type " + uf.getFileName());
+        }
+    }
+
+    private boolean isAllowedType(String fileName) {
+        String allowTypes = getAllowTypes();
+        return allowTypes == null || allowTypes.length() < 3 || // JS RegExp has at least 3 chars
+                Pattern.compile(allowTypes.substring(1, allowTypes.length() - 1))
+                .matcher(fileName).find();
+    }
+
+    private String getAllowTypes() {
+        // some tests break in Mojarra when binding is used
+        // search for first FileUpload component in component tree (assuming this is the only one)
+        if (_allowTypes == null) {
```

Comment:
wouldnt it be better to use findComponent by id?

Reply:
I do not have a clear opinion. All FileUpload tests are backed by the same two beans `FileUploadHandler` and `FileUploadView` At the moment all xhtml files contain a single p:fileUpload component which do have all the same id="form:fileupload". So yes, findComponent can be used.
Because it's less and simpler code I'll change that to findComponent.